### ISSUE TITLE
Prq should have different semantic than Req in erase job

### ISSUE THREAD

**Issue by jsilhan** [openSUSE/libsolv#126]
Hi,
when you uninstall the dependency of some package then the package should be taken away too. This behaves as expected:
```
repo system 0 testtags <inline>
#>=Pkg: A 1 1 x86_64
#>=Req: B
#>=Pkg: B 1 1 x86_64
repo available 0 testtags <inline>
system x86_64 rpm system
poolflags implicitobsoleteusescolors
solverflags allowvendorchange allowuninstall keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
job erase name B
result transaction,problems <inline>
#>erase A-1-1.x86_64@system
#>erase B-1-1.x86_64@system
```

When you have a `Requires(pre)` then the "required" package is not actually needed for package proper run. So what I would expect in following testcase is this:
```
repo system 0 testtags <inline>
#>=Pkg: A 1 1 x86_64
#>=Prq: B
#>=Pkg: B 1 1 x86_64
repo available 0 testtags <inline>
system x86_64 rpm system
poolflags implicitobsoleteusescolors
solverflags allowvendorchange allowuninstall keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
job erase name B
result transaction,problems <inline>
#>erase B-1-1.x86_64@system
```
but it erases package `A` too.

https://bugzilla.redhat.com/show_bug.cgi?id=1292915
https://bugzilla.redhat.com/show_bug.cgi?id=1310840

FYI when `cleandeps` is specified in `erase` job,  `Prq` acts like `Req` too (which is probably right but `Prq` could be ignored too). I this case we don't know how it should act. We don't mind.
```
repo system 0 testtags <inline>
#>=Pkg: A 1 1 x86_64
#>=Pkg: B 1 1 x86_64
#>=Prq: A
repo available 0 testtags <inline>
system x86_64 rpm system
poolflags implicitobsoleteusescolors
solverflags allowvendorchange allowuninstall keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
job erase name B [cleandeps]
result transaction,problems <inline>
erase A-1-1.x86_64@system
erase B-1-1.x86_64@system
```

**Comment by mlschroe** [comment]
Yes, it could leave A behind. To change this, we would need to add the repo_add_rpmdb function to treat Requires(pre) differently. But then people may complain that the requires get changed if a package is installed, which is also not so nice.

**Comment by mlschroe** [comment]
Now fixed in libsolv-0.6.20. I added a way to specify which pre-deps should be ignored for installed packages. You need to adapt your testcases if you want to test this, there's a new 'Ipr' tag.

repo_add_rpmdb automatically adds to needed data, so no code change is needed. Hmm. Except that I should have changed the RPMDB_COOKIE_VERSION in repo_rpmdb.c as well. I'll commit that on Monday.
